esp-hal/src/rtc_cntl/sleep/esp32s2.rs

```diff
@@ -0,0 +1,811 @@
+                // power mode during light sleep
+                rtc_sleep_pu(false);
+
+                rtc_cntl.dig_pwc().modify(|_, w| {
+                    w.dg_wrap_force_pu().clear_bit();
+                    w.wifi_force_pu().clear_bit()
+                });
+
+                rtc_cntl.dig_iso().modify(|_, w| {
+                    w.dg_wrap_force_noiso().clear_bit();
+                    // NOTE: not present in idf.
+                    w.dg_wrap_force_iso().clear_bit()
+                });
+
+                rtc_cntl.dig_iso().modify(|_, w| {
+                    w.wifi_force_noiso().clear_bit();
+                    // NOTE: not present in idf.
+                    w.wifi_force_iso().clear_bit()
+                });
+
+                rtc_cntl.pwc().modify(|_, w| w.force_noiso().clear_bit());
+
+                // cancel digital PADS force no iso
+                system
+                    .cpu_per_conf()
+                    .modify(|_, w| w.cpu_wait_mode_force_on().clear_bit());
+
+                // if DPORT_CPU_WAIT_MODE_FORCE_ON == 0,
+                // the cpu clk will be closed when cpu enter WAITI mode
+                rtc_cntl.dig_iso().modify(|_, w| {
+                    w.dg_pad_force_unhold().clear_bit();
+                    w.dg_pad_force_noiso().clear_bit()
+                });
+            }
+
+            // force power down wifi and bt power domain
+            rtc_cntl
+                .dig_iso()
+                .modify(|_, w| w.wifi_force_iso().set_bit());
+
+            rtc_cntl
+                .dig_pwc()
+                .modify(|_, w| w.wifi_force_pd().set_bit());
+
+            rtc_cntl.int_ena().write(|w| w.bits(0));
+            rtc_cntl.int_clr().write(|w| w.bits(u32::MAX));
+        }
+    }
+
+    pub(crate) fn apply(&self) {
+        // like esp-idf rtc_sleep_init() and deep_sleep_start()
+        let rtc_cntl = LPWR::regs();
+
+        if self.deep_slp() {
+            // "Due to hardware limitations, on S2 the brownout detector
+            // sometimes trigger during deep sleep to circumvent
+            // this we disable the brownout detector before sleeping' - from
+            // idf's deep_sleep_start()
+            unsafe {
+                // brownout_hal_config(brownlout_hal_config_t{0})
+                rtc_cntl.brown_out().modify(|_, w| {
+                    w.int_wait().bits(2);
+                    w.close_flash_ena().clear_bit();
+                    w.pd_rf_ena().clear_bit();
+                    w.cnt_clr().set_bit()
+                });
+                rtc_cntl.brown_out().modify(|_, w| {
+                    // Set followed by clear in idf
+                    w.cnt_clr().clear_bit();
+                    w.rst_wait().bits(0x3fff);
+                    w.rst_ena().clear_bit();
+                    w.brown_out2_ena().set_bit();
+                    w.rst_sel().set_bit()
+                });
+                regi2c::I2C_BOD_REG_THRESHOLD.write_field(0);
```

Comment:
So while I think the right thing is to port over the regi2c read function, this is likely the bit that eventually triggers the build error. I think we should be safe to just remove any of the code above `ena().clear_bit()` - there's no point in configuring the brownout detector if we are then disabling it.